### PR TITLE
Source predicate can be by passed in AWS

### PR DESCRIPTION
**Pull request by szuecs**
elb and alb and probably will append all X-Forwarded-For header items they get in the request header to their own in the request they do to the backend. As result the right most is the real client IP.

TODO:
- [x] add an option to select left or right most matching in Source() predicate or skipper cli
- [x] do not change the default to not break implementations based on this

see https://github.com/zalando/skipper/issues/531#issuecomment-355320512

### CODE REVIEW

**Comment by lmineiro** [comment]
Is this a breaking change?

**Comment by szuecs** [comment]
I changed the description of the PR to make sure it is understood what is open todo to fix the issue, before merge

**Comment by lmineiro** [comment]
For reference, since there were discussions around Skipper's proxy behavior in the past:
* https://github.com/zalando/skipper/issues/61
* https://github.com/zalando/skipper/issues/250

**Comment by aryszka** [comment]
how about using a new predicate name for this behavior: e.g. LastSource() or s.g. like that?

**Comment by lmineiro** [comment]
Or `CloserSource`?

**Comment by szuecs** [comment]
@aryszka I am fine with that, but we have to pass an option for running in AWS anyways, because otherwise we hardcode the AWS way for us in the dataclient which does not makes sense.

@lmineiro CloserSource ? What? Why?

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
@aryszka I created a second predicate with tests and an option to skipper that can be used to change the automatic healthz endpoint to use the not common AWS way of X-Forwarded-For header to find the client ip.

File: net/net.go [hunk 1]
```diff
@@ -17,13 +17,20 @@ func stripPort(address string) string {
...
+		return res
 	}
 	return nil
 }
 
-// The remote address of the client. When the 'X-Forwarded-For'
-// header is set, then it is used instead.
+// RemoteHost returns the remote address of the client. When the
+// 'X-Forwarded-For' header is set, then it is used instead. This is
+// the most often behaviour for proxies. Wikipedia shows the format
```
Comment (aryszka) [inline net/net.go]: "This is how most often proxies behave."

File: predicates/source/source.go [hunk 2]
```diff
@@ -86,11 +99,51 @@ func (s *spec) Create(args []interface{}) (routing.Predicate, error) {
...
 }
+
+func NewFromLast() routing.PredicateSpec { return &specLast{} }
+
+func (s *specLast) Name() string {
+	return Name
+}
+
+func (s *specLast) Create(args []interface{}) (routing.Predicate, error) {
+	if len(args) == 0 {
```
Comment (aryszka) [inline predicates/source/source.go]: this duplicate code could be avoided, if the spec would be the same for the two flavors with the bool field passed to the predicate instance. The separate New() and NewFromLast() functions should stay but return the same spec with this bool field difference.

**Comment by aryszka** [comment]
@szuecs looks good to me. I only found that the two flavors of the predicate could share the Create() method if the spec had a bool field like fromLast

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**